### PR TITLE
flamenco: status cache in runtime

### PR DESCRIPTION
**Pull request by arjain4**
Runtime changes to use status cache

### CODE REVIEW

File: src/flamenco/snapshot/fd_snapshot_restore.h [hunk 1]
```diff
@@ -81,8 +98,10 @@ fd_snapshot_restore_new( void *                               mem,
                          fd_acc_mgr_t *                       acc_mgr,
                          fd_funk_txn_t *                      txn,
                          fd_valloc_t                          valloc,
-                         void *                               cb_ctx,
-                         fd_snapshot_restore_cb_manifest_fn_t cb );
+                         void *                               cb_manifest_ctx,
+                         fd_snapshot_restore_cb_manifest_fn_t cb_manifest,
+                         void *                               cb_status_cache_ctx,
+                         fd_snapshot_restore_cb_status_cache_fn_t cb_status_cache );
```
Comment (ripatel-fd) [inline src/flamenco/snapshot/fd_snapshot_restore.h:104]: The callback context would typically be the same
```suggestion
                         void *                               cb_ctx,
                         fd_snapshot_restore_cb_manifest_fn_t cb_manifest,
                         fd_snapshot_restore_cb_status_cache_fn_t cb_status_cache );
```

File: src/flamenco/snapshot/test_snapshot_restore.c
Comment (ripatel-fd) [inline src/flamenco/snapshot/test_snapshot_restore.c:1]: Before merging, please add tests here

File: src/flamenco/runtime/fd_runtime.c [hunk 2]
```diff
@@ -876,6 +876,23 @@ fd_runtime_prepare_txns_phase2_tpool( fd_exec_slot_ctx_t * slot_ctx,
         }
       }
 
+      fd_txncache_query_t curr_query;
+      curr_query.blockhash = blockhash->uc;
+      fd_blake3_t b3[1];
+      uchar hash[32];
+      fd_blake3_init( b3 );
+      fd_blake3_append( b3, ((uchar *)txn_ctx->_txn_raw->raw + txn_ctx->txn_descriptor->message_off), txn_ctx->_txn_raw->txn_sz - txn_ctx->txn_descriptor->message_off );
+      fd_blake3_fini( b3, hash );
```
Comment (ripatel-fd) [inline src/flamenco/runtime/fd_runtime.c:885]: Prefer fd_blake3_hash. The streamlined function is faster in the Blake3 rewrite

File: src/flamenco/snapshot/fd_snapshot.c
Comment (ripatel-fd) [inline src/flamenco/snapshot/fd_snapshot.c:1]: Snapshots are always at rooted slots. What is the purpose of recovering the status cache?

Comment (arjain4) [reply src/flamenco/snapshot/fd_snapshot.c:1]: We have to store the last 300 rooted slots in the status cache. Being rooted doesn't imply evicted from the status cache

File: src/flamenco/runtime/fd_runtime.c (hunk 2 above)
Comment (arjain4) [reply src/flamenco/runtime/fd_runtime.c:885]: that api isn't available yet

**Review by lheeger-jump — DISMISSED** [review]

**Review by lheeger-jump — APPROVED** [review]

**Pull request merged**